### PR TITLE
Update Shapes How-To for new Lasso tool (napari/#5555)

### PR DESCRIPTION
**Pull request by melonora**
# Description
This PR adds documentation for the lasso tool that would soon be added to Napari [#5555](https://github.com/napari/napari/pull/5555)

## Type of change

- [x] Fixes or improves existing content
- [ ] Adds new content page(s)
- [ ] Fixes or improves workflow, documentation build or deployment

# References
depends on https://github.com/napari/napari/pull/5555

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added [alt text](https://webaim.org/techniques/alttext/) to new images included in this PR

### CODE REVIEW

File: docs/howtos/layers/shapes.md [hunk 1]
```diff
@@ -140,12 +140,38 @@ centered on that click.
...
-complete the polygon. You'll then be able to start adding another one.
+polygon, which can be non-convex, but cannot have holes. For drawing polygons, 
+multiple tools can be used.
+The polygon creation tool can be selected from the layer control panel or by 
+pressing the `P` key when the shapes layer is selected. When adding a polygon 
+each click will add a vertex at the clicked location. To finish drawing a polygon 
+you must click the`escape` key, which will add a final vertex at the current mouse 
+position and complete the polygon. You'll then be able to start adding another one.
+
+`Polygons` can also be created with the polygon lasso creation tool. This can be found
```
Comment (psobolewskiPhD) [inline docs/howtos/layers/shapes.md]:
```suggestion
`Polygons` can also be created with the polygon lasso creation tool, which can be found
```

File: docs/howtos/layers/shapes.md [hunk 2]
```diff
@@ -140,12 +140,38 @@ centered on that click.
...
+in the layer control panel or by pressing `shift+P`. The tool can be used to draw 
+complex `Polygons` with the mouse
```
Comment (psobolewskiPhD) [inline docs/howtos/layers/shapes.md]:
```suggestion
complex `Polygons` with the mouse.
```

File: docs/howtos/layers/shapes.md [hunk 3]
```diff
@@ -140,12 +140,38 @@ centered on that click.
...
+For mouse draw mode the sequence of events by the user is as follows:
```
Comment (psobolewskiPhD) [inline docs/howtos/layers/shapes.md]:
```suggestion
The polygon lasso supports using the mouse or tablet. When using the mouse, the sequence of events to draw a polygon is as follows:
```

File: docs/howtos/layers/shapes.md [hunk 4]
```diff
@@ -140,12 +140,38 @@ centered on that click.
...
+
+    1. Mouse press
+    2. Move mouse while not mouse pressed
+    3. Mouse press to finish draw
```
Comment (psobolewskiPhD) [inline docs/howtos/layers/shapes.md]:
```suggestion
    1. Click mouse (left-click) to begin drawing.
    2. Move mouse—without holding down the mouse button—to draw the polygon.
    3. Click mouse (left-click) or press `escape` to end drawing—the polygon will auto-complete.
```

File: docs/howtos/layers/shapes.md [hunk 5]
```diff
@@ -140,12 +140,38 @@ centered on that click.
...
+
+`Polygons` can also be drawn using a tablet when using the polygon lasso tool. In this case,
```
Comment (psobolewskiPhD) [inline docs/howtos/layers/shapes.md]:
```suggestion
The polygon lasso tool can also be used to draw `Polygons` using a tablet. In this case, 
```

File: docs/howtos/layers/shapes.md [hunk 6]
```diff
@@ -140,12 +140,38 @@ centered on that click.
...
+complex `Polygons` with the mouse
+For mouse draw mode the sequence of events by the user is as follows:
+
+    1. Mouse press
+    2. Move mouse while not mouse pressed
+    3. Mouse press to finish draw
+
+`Polygons` can also be drawn using a tablet when using the polygon lasso tool. In this case, 
+drawing is started by touching the tablet screen with the tablet pencil. The drawing will 
+be ongoing for as long as the pencil is moved while touching the tablet screen. For both
```
Comment (psobolewskiPhD) [inline docs/howtos/layers/shapes.md]:
```suggestion
drawing the polygon is started by touching the tablet screen with the tablet stylus and drawing will continue for as long as the pencil is moved while touching the tablet screen. Note that similar behavior is also available when using a macOS trackpad, using three-finger drag mode.  
For both 
```

File: docs/howtos/layers/shapes.md [hunk 7]
```diff
@@ -140,12 +140,38 @@ centered on that click.
...
+
+`Polygons` can also be drawn using a tablet when using the polygon lasso tool. In this case, 
+drawing is started by touching the tablet screen with the tablet pencil. The drawing will 
+be ongoing for as long as the pencil is moved while touching the tablet screen. For both 
+modes, vertices are added only if the vertex to be added is at least 10 screen pixels away 
+from the previous vertex. As with the polygon creation tool drawing the shape can also be 
+finished by pressing the `escape` key.
+
+When finishing a drawing using the polygon lasso tool an implementation of the Ramer–Douglas–Peucker 
+algorithm is applied to reduce the number of points that make up the shape, while conserving its
```
Comment (psobolewskiPhD) [inline docs/howtos/layers/shapes.md]:
```suggestion
After finishing drawing a polygon using the polygon lasso tool, an implementation of the [Ramer–Douglas–Peucker 
algorithm](https://en.wikipedia.org/wiki/Ramer–Douglas–Peucker_algorithm) is applied to reduce the number of vertexes that make up the shape, while preserving its
```

File: docs/howtos/layers/shapes.md [hunk 8]
```diff
@@ -140,12 +140,38 @@ centered on that click.
...
+contours. The agressiveness by which the algorithm reduces the number of points of a shape is
```
Comment (psobolewskiPhD) [inline docs/howtos/layers/shapes.md]:
```suggestion
contours. The aggressiveness with which the algorithm reduces the number of points of the polygon is 
```

File: docs/howtos/layers/shapes.md [hunk 9]
```diff
@@ -140,12 +140,38 @@ centered on that click.
...
+be ongoing for as long as the pencil is moved while touching the tablet screen. For both 
+modes, vertices are added only if the vertex to be added is at least 10 screen pixels away 
+from the previous vertex. As with the polygon creation tool drawing the shape can also be 
+finished by pressing the `escape` key.
+
+When finishing a drawing using the polygon lasso tool an implementation of the Ramer–Douglas–Peucker 
+algorithm is applied to reduce the number of points that make up the shape, while conserving its
+contours. The agressiveness by which the algorithm reduces the number of points of a shape is 
+determined by an `epsilon` parameter. If 0 no points are removed. The higher the value of `epsilon`
+the more points get removed. The value of `epsilon` can be set by in napari going to
```
Comment (psobolewskiPhD) [inline docs/howtos/layers/shapes.md]:
```suggestion
determined by an `epsilon` parameter, which is a perpendicular distance threshold. Any vertexes beyond the threshold will be preserved, so if `epsilon` is set to `0`, no vertexes will be removed. With increasing values of `epsilon`, more and more vertexes will be removed. The value of `epsilon` can be set by in napari going to 
```

**Review by psobolewskiPhD — CHANGES_REQUESTED** [review]
I made a few suggestions/clarifications.
I suggest a link to wikipedia for the RDP, but if you have some other resource instead, please feel free to edit.
Also, I edited the OP to `depend` on the napari PR to ensure we don't merge this prematurely.

File: docs/howtos/layers/shapes.md (hunk 1 above)
Comment (melonora) [reply docs/howtos/layers/shapes.md]: Originally I had it the same, but wasn't certain about creating longer sentences.

File: docs/howtos/layers/shapes.md [hunk 10]
```diff
@@ -140,12 +140,43 @@ centered on that click.
...
+multiple tools can be used.
+The polygon creation tool can be selected from the layer control panel or by 
+pressing the `P` key when the shapes layer is selected. When adding a polygon 
+each click will add a vertex at the clicked location. To finish drawing a polygon 
+you must click the`escape` key, which will add a final vertex at the current mouse 
+position and complete the polygon. You'll then be able to start adding another one.
+
+`Polygons` can also be created with the polygon lasso creation tool, which can be 
+found in the layer control panel or by pressing `shift+P`. The tool can be used to 
+draw complex `Polygons` with the mouse. The polygon lasso supports using the mouse
```
Comment (psobolewskiPhD) [inline docs/howtos/layers/shapes.md]:
```suggestion
draw complex `Polygons` with the mouse
```

File: docs/howtos/layers/shapes.md [hunk 11]
```diff
@@ -140,12 +140,43 @@ centered on that click.
...
+
+`Polygons` can also be created with the polygon lasso creation tool, which can be 
+found in the layer control panel or by pressing `shift+P`. The tool can be used to 
+draw complex `Polygons` with the mouse. The polygon lasso supports using the mouse 
+or tablet. When using the mouse, the sequence of events to draw a polygon is as 
+follows:
+
+    1. Click mouse (left-click) to begin drawing.
+    2. Move mouse — without holding down the mouse button — to draw the polygon.
+    3. Click mouse (left-click) or press `escape` to end drawing—the polygon will auto-complete.
```
Comment (psobolewskiPhD) [inline docs/howtos/layers/shapes.md]:
```suggestion
1. Click mouse (left-click) to begin drawing.
2. Move mouse — without holding down the mouse button — to draw the polygon.
3. Click mouse (left-click) or press `escape` to end drawing—the polygon will auto-complete.
```

With the indentation this is showing as a code block and looks kinda bad:
image

File: docs/howtos/layers/shapes.md [hunk 12]
```diff
@@ -140,12 +140,43 @@ centered on that click.
...
+Note that similar behavior is also available when using a macOS trackpad, using three-finger 
+drag mode.  
+For both modes, vertices are added only if the vertex to be added is at least 10 screen pixels 
+away from the previous vertex. As with the polygon creation tool drawing the shape can also be 
+finished by pressing the `escape` key.
+
+After finishing drawing a polygon using the polygon lasso tool, an implementation of the [Ramer–Douglas–Peucker 
+algorithm](https://en.wikipedia.org/wiki/Ramer–Douglas–Peucker_algorithm) is applied to reduce the 
+number of vertices that make up the shape, while preserving its contours. The aggressiveness with 
+which the algorithm reduces the number of points of the polygon is determined by an `epsilon` parameter,
```
Comment (psobolewskiPhD) [inline docs/howtos/layers/shapes.md]:
```suggestion
which the algorithm reduces the number of vertices of the polygon is determined by an `epsilon` parameter, 
```

**Review by psobolewskiPhD — APPROVED** [review]
Couple more minor nitpicks, but seems clear to me!
Thanks!

**Comment by psobolewskiPhD** [comment]
The tool PR: https://github.com/napari/napari/pull/5555 has 0.4.18 milestone.

**Pull request merged**